app/routes/questionnaire.py

```diff
@@ -276,6 +287,39 @@ def get_thank_you(schema):
     )
 
 
+@post_submission_blueprint.route("email/another/", methods=["GET", "POST"])
+@login_required
+def get_another_email():
+    another_email = AnotherEmail()
```

Comment:
The `AnotherEmail` and `ThankYou` class have very similar logic, the Email logic could be made independent to reduce the duplication in `ThankYou` class. From what I can see, the only difference is thank you's `validate()` saves the session which isn't ideal either given the method is called `validate`. If both endpoints used the same class, on a POST to thank you or another email endpoint, you could instantiate the same class and reuse the same logic.

Reply:
Lost me on this one, the init is different as Thank you has to check the schema; the context build has to be different as well as thank you needs the context for thank you; the init for anotheremail has to check the session to see if it is reachable and thank you needs to persist that value